cmake/mongocrypt-config.cmake

```diff
@@ -1,3 +1,58 @@
 include(CMakeFindDependencyMacro)
 find_dependency(kms_message 0.0.1)
 include("${CMAKE_CURRENT_LIST_DIR}/mongocrypt_targets.cmake")
+
+# Find the libbson that will be used for static linking
+if (DEFINED MONGOCRYPT_LIBBSON_STATIC_USE)
```

Comment:
Is there a use case for consumers defining `MONGOCRYPT_LIBBSON_STATIC_USE` or `MONGOCRYPT_LIBBSON_SHARED_USE`. Does this override the target names?